### ISSUE TITLE
Missing years in map charts

### ISSUE THREAD

**Issue by AFoletti** [visualize-admin/visualization-tool#852]
**Describe the bug**
When using a map chart for [this cube](https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/5), only some years (2014 and 2015) are available. The full range (2014 to 2021) is available when using e.g. a bar chart.
**This issue is on PROD too**, visible [here](https://www.uvek-gis.admin.ch/BFE/storymaps/EE_Einmalverguetung/).

**To Reproduce**
1. Go to [visualize](https://int.visualize.admin.ch/) and look for the "Einmalvergütung für Photovoltaikanlagen" dataset (filter by BFE data if needed)
2. Open the dataset for visualization. Devine a bar chart using "Jahr der Vergütung" for the horizontal axis. Years 2014 to 2021 are displayed. (OK)
3. Now switch to map chart.
4. Only years 2014 and 2015 are available as filter years (NOK)

**Expected behavior**
Every year is available as filter

**Screenshots**

**Environment (please complete the following information):**
 - Visualize environment and version: v3.12.0 ([c7820ce](https://github.com/visualize-admin/visualization-tool/commit/c7820ce5fa02ab4dea39b2c64b6ebcc265b11f49)) using PROD data, and v3.10.0 ([13b8295](https://github.com/visualize-admin/visualization-tool/commit/13b8295589020d7817ba2fbc10e1f96da4d192fd)) too
 - Browser and version: Edge

**Additional context**
Since this is a PROD issue too, and appeared recently, I guess it's somehow linked to a recend change I made to the cube following a request from @bprusinowski in late October. Following his remarks I switched two measured for this cube to "interval" (check differences between [version 4](https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/4) and [version 5](https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/5) of the cube, if needed).
Nothing else was changed and the filter worked fine in the past.

**Comment by bprusinowski** [comment]
Hi @AFoletti, thanks for reporting this issue. It seems it's not related to the maps only, the same problem appears in other charts (when you e.g. change X dimension from Year to Kanton).

In this particular case it's related to the fact that Year dimension has an Ordinal scale type, while for dates it should be Interval. Changing this should fix the issue on all environments; would this be possible?

I also introduced a change in #854 to allow correct parsing on Temporal dimensions with scale type different than Interval to prevent this from happening (also, a new cube checker rule has been added there, to make the information about correct scale type for Temporal dimensions more accessible).

**Comment by AFoletti** [comment]
hey @bprusinowski 
Thanks for your feedback. I did guess the problem was in the data. Nevertheless, I did not change the Year scale type and it was working fine some time ago. There is then another secondary cause which is not identified yet. Up to you if you want to investigate it further.
On my side, the viz are now working fine again with the new scale type.
Moreover, I appreciate the check in #854. I think that from a data quality perspective, the cube creator should check for "Time description/Year - scale type" combination and disallow the publication if found wrong.
